src/comments.js

```diff
@@ -425,6 +445,35 @@ function printComment(commentPath) {
   }
 }
 
+function findExpressionIndexForComment (expressions, comment, options) {
+  var match;
+  const startPos = locStart(comment) - 1;
+  const endPos = locEnd(comment) + 1;
+
+  for (var i = 0; i < expressions.length; ++i) {
```

Comment:
Can you make a function

```js
getExpressionRange(expr) {
  if (expr.start) {
    // Babylon
    return {start: expr.start, end: expr.end}
  }
  // Flow
  return {start: expr.range[0], end: expr.range[1]}
}
```

this way you don't have to thread the parser everywhere.